### ISSUE TITLE
ObjectPropertyNaming should to flag const properties that are not simple types

### ISSUE THREAD

**Issue by kenyee** [arturbosch/detekt#1167]
Your issue may already be reported!
Please search in the [issues](https://github.com/arturbosch/detekt/issues) before creating one.

## Expected Behavior
ObjectPropertyNaming rule should not indicate false positives

## Current Behavior
In a companion object, Kotlin doesn't allow you to make a property const unless it's a simple data type (string, int, etc.).
However, the ObjectPropertyNaming rule will flag naming for values like this:
private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMdd-HH:mm")

## Steps to Reproduce (for bugs)
add above to a companion object and run detekt against it

I tried using the default detekt.yml config for this rule:

```
   constantPattern: '[A-Za-z][_A-Za-z0-9]*'
   propertyPattern: '[A-Za-z][_A-Za-z0-9]*'
   privatePropertyPattern: '(_)?[A-Za-z][A-Za-z0-9]*'
```
as well as using this suggested in the #detekt slack:

```
   privatePropertyPattern: '[A-Za-z][A-Za-z0-9]*'
```

but still get false positives

## Context
This is a useful rule that mirrors Sonar's rule, but the false positives make it not useful

## Your Environment
* Version used: Detekt 1.0.0 RC9.2
* Operating System and version: OSX latest

**Comment by arturbosch** [comment]
I think I don't quite get what you mean.
Testen with:
```

		it("#1167") {
			val code = """
				class A {
					companion object {
						private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMdd-HH:mm")
					}
				}
			""".trimIndent()
			assertThat(subject.lint(code)).isEmpty()
		}
``` It looks alright to me?

**Comment by kenyee** [comment]
How do you have your detekt.yml configured for the matchers?  
Mine is:
```
  ObjectPropertyNaming:
    active: true
    constantPattern: '[A-Za-z][_A-Za-z0-9]*'
    propertyPattern: '[A-Za-z][_A-Za-z0-9]*'
    privatePropertyPattern: '[A-Za-z][A-Za-z0-9]*'
```

Also doesn't need to be in a companion object.  Can just be in a singleton like this and it gets flagged erroneously by the rule:
```
object Utils {
  private val USER_CURRENCY_OVERRIDE_LIMIT_PERCENTAGE = BigDecimal.valueOf(0.1f)
}
```

I think it's because it doesn't know whether a property is a primitive vs. an object.  Kotlin won't allow you to specify that the above is a const val.  The rule's matchers appear to be just regexes...

**Comment by schalkms** [comment]
Detekt complies with the Kotlin style guide. 
Your sample `private val DATE_TIME_FORMATTER = ...` is not reported if you configure it in the `privatePropertyPattern` option.
